lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1627,6 +1627,38 @@ static OpFoldResult atenBinaryIntOperatorFoldHelper(OpTy op,
   return getI64IntegerAttr(op.getContext(), f(lhs, rhs));
 }
 
+//===----------------------------------------------------------------------===//
+// AtenFindOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenFindOp::fold(ArrayRef<Attribute> operands) {
+  int64_t startConstant, endConstant;
+  if (!matchPattern(start(), m_TorchConstantInt(&startConstant)) ||
+      !matchPattern(end(), m_TorchConstantInt(&endConstant)))
+    return nullptr;
+
+  std::string selfStr, targetStr;
+  if (!matchPattern(self(), m_TorchConstantStr(selfStr)) ||
+      !matchPattern(substr(), m_TorchConstantStr(targetStr)))
+    return nullptr;
+
+  int64_t selfLen = selfStr.size();
+  startConstant = startConstant < 0 ? selfLen + startConstant : startConstant;
+  endConstant = endConstant < 0 ? selfLen + endConstant + 1 : endConstant;
+
+  if (startConstant >= endConstant) {
+    if (targetStr.size() == 0)
+      return getI64IntegerAttr(getContext(), startConstant);
+    return getI64IntegerAttr(getContext(), -1);
+  }
```

Comment:
I think you've found a bug in the `torch.ops.aten.find` op! The behavior of the op is not consistent with python's `str.find`:

```python
In [4]: torch.ops.aten.find("Hello", "", 900, 5)
Out[4]: 900

In [5]: "Hello".find("", 900, 5)
Out[5]: -1
```

Can you file a bug in upstream pytorch with this finding?

Reply:
This did seem like strange behavior; I will file the issue upstream then and table this PR for now.